components/launcher/src/server.rs

```diff
@@ -271,12 +271,11 @@ impl Server {
                     return result;
                 }
             }
-            Some(SignalEvent::Passthrough(signal)) => {
-                self.forward_signal(signal);
+
+            if signals::pending_sighup() {
```

Comment:
_Tecnically_ this is a small behavior change for the launcher.  Whereas before we would only handle 1 signal on each tick, now we potentially handle two.  If we are concerned we could move this to an `if else`.